ArcGISRuntimeSDKQt_QMLSamples/UtilityNetwork/PerformValveIsolationTrace/PerformValveIsolationTrace.qml

```diff
@@ -26,22 +26,86 @@ Rectangle {
     width: 800
     height: 600
 
-    readonly property url featureServiceUrl: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleGas/FeatureServer"
+    readonly property url featureServiceUrl: "https://sampleserver7.arcgisonline.com/server/rest/services/UtilityNetwork/NapervilleGas/FeatureServer"
     readonly property string domainNetworkName: "Pipeline"
     readonly property string tierName: "Pipe Distribution System"
     readonly property string networkSourceName: "Gas Device"
     readonly property string assetGroupName: "Meter"
     readonly property string assetTypeName: "Customer"
     readonly property string globalId: "98A06E95-70BE-43E7-91B7-E34C9D3CB9FF"
+    readonly property string sampleServer7Username: "viewer01"
+    readonly property string sampleServer7Password: "I68VGU^nMurF"
+    property var clickPoint: null
+    property var filterBarriers: []
+    property var terminalNames: []
+    property var createdElement: null
     property var traceConfiguration: null
     property var startingLocation: null
     property var categories: null
     property bool uiEnabled: false
+    property var distributionLineLayerFeatureTable: null
+    property var deviceLayerFeatureTable: null
 
     MapView {
         id: mapView
         anchors.fill: parent
 
+        onMouseClicked: {
+            const screenX = mouse.x;
+            const screenY = mouse.y;
+            const tolerance = 10;
+            const returnPopups = false;
+            clickPoint = mouse.mapPoint;
+            mapView.identifyLayers(screenX, screenY, tolerance, returnPopups);
+        }
+
+        onIdentifyLayersStatusChanged : {
+            if (identifyLayersStatus !== Enums.TaskStatusCompleted)
+                return;
+
+            const results = mapView.identifyLayersResults;
+            const identifiedLayerResult = results[0];
+
+            // could not identify location
+            if (!identifiedLayerResult)
+                return;
+
+            const feature = identifiedLayerResult.geoElements[0];
+
+            let element = utilityNetwork.createElementWithArcGISFeature(feature);
+
+            createdElement = element;
+
+            const elementSourceType = element.networkSource.sourceType;
+
+            if (elementSourceType === Enums.UtilityNetworkSourceTypeJunction) {
+                const terminals = element.assetType.terminalConfiguration.terminals;
+                if (terminals.length > 1) {
+                    terminalNames = [];
+                    const temporaryTerminalNames = [];
```

Comment:
Just curious why you used a new variable here instead of directly pushing to `terminalNames`?

Reply:
Ah I was talking to @JamesMBallard about this briefly and I forgot to double back on this. Some reason when just pushing the the global variable the `TerminalPickerView` wasn't recognizing the list was updated. This was just a temporary solution.

Reply:
Switched to a declared ListModel instead of a list property being used as a model.